lib/src/views/Calendar/CalendarHeader.tsx

```diff
@@ -1,124 +1,178 @@
 import * as React from 'react';
 import * as PropTypes from 'prop-types';
+import clsx from 'clsx';
 import Typography from '@material-ui/core/Typography';
-import SlideTransition, { SlideDirection } from './SlideTransition';
-import IconButton, { IconButtonProps } from '@material-ui/core/IconButton';
-import { DateType } from '@date-io/type';
+import { CalendarProps } from './Calendar';
+import { DatePickerView } from '../../DatePicker';
+import { SlideDirection } from './SlideTransition';
+import { Fade, IconButton } from '@material-ui/core';
 import { useUtils } from '../../_shared/hooks/useUtils';
+import { VariantContext } from '../../wrappers/Wrapper';
 import { MaterialUiPickersDate } from '../../typings/date';
-import { makeStyles, useTheme } from '@material-ui/core/styles';
+import { FadeTransitionGroup } from './FadeTransitionGroup';
+import { IconButtonProps } from '@material-ui/core/IconButton';
 import { ArrowLeftIcon } from '../../_shared/icons/ArrowLeftIcon';
 import { ArrowRightIcon } from '../../_shared/icons/ArrowRightIcon';
+import { makeStyles, useTheme, Theme } from '@material-ui/core/styles';
+import { ArrowDropDownIcon } from '../../_shared/icons/ArrowDropDownIcon';
 
-export interface CalendarHeaderProps {
-  currentMonth: DateType;
+export interface CalendarWithHeaderProps
+  extends Pick<CalendarProps, 'minDate' | 'maxDate' | 'disablePast' | 'disableFuture'> {
+  view: DatePickerView;
+  month: MaterialUiPickersDate;
+  /** Left arrow icon */
   leftArrowIcon?: React.ReactNode;
+  /** Right arrow icon */
   rightArrowIcon?: React.ReactNode;
+  /**
+   * Props to pass to left arrow button
+   * @type {Partial<IconButtonProps>}
+   */
   leftArrowButtonProps?: Partial<IconButtonProps>;
+  /**
+   * Props to pass to right arrow button
+   * @type {Partial<IconButtonProps>}
+   */
   rightArrowButtonProps?: Partial<IconButtonProps>;
-  disablePrevMonth?: boolean;
-  disableNextMonth?: boolean;
-  slideDirection: SlideDirection;
-  onMonthChange: (date: MaterialUiPickersDate, direction: SlideDirection) => void | Promise<void>;
+  changeView: () => void;
+  onMonthChange: (date: MaterialUiPickersDate, slideDirection: SlideDirection) => void;
 }
 
-export const useStyles = makeStyles(
+export const useStyles = makeStyles<Theme, Pick<CalendarWithHeaderProps, 'view'>>(
   theme => ({
     switchHeader: {
       display: 'flex',
-      justifyContent: 'space-between',
       alignItems: 'center',
-      marginTop: theme.spacing(0.5),
+      marginTop: theme.spacing(2),
       marginBottom: theme.spacing(1),
+      paddingLeft: theme.spacing(3),
```

Comment:
I would suggest not to use the spacing helper in these components unless if somebody customizes it, it has an expected outcome.

Reply:
@oliviertassinari it is a nice point. I am really not clear about what spacing means. But we have some issues like #1141. How do you think should we support them or just ignore?

Reply:
Yes, #1128 is exactly why I would discourage using the spacing helper unless a different have a clear expected outcome (we have very few cases on this on the main repository). I can give you one example `<Grid spacing={1} />`. I doubt there are any cases like this in the date picker.